src/senaite/core/upgrade/v02_05_000.py

```diff
@@ -380,3 +383,55 @@ def remove_legacy_reports(tool):
     portal.portal_types.manage_delObjects(["Report", "ReportFolder"])
 
     logger.info("Removing legacy reports [DONE]")
+
+
+def purge_catalogs(tool):
+    """Purges stale indexes and metadata from catalogs
+    """
+    logger.info("Purging catalogs ...")
+
+    indexes_to_remove = {
+        "base_catalog": [
+            "CreationDate",
+        ],
+        SAMPLE_CATALOG: [
+            "getClientID",
+            "getClientUID",
+        ],
+        CLIENT_CATALOG: [
+            "getClientID",
+        ],
+        SENAITE_CATALOG: [
+            "getClientID"
+        ],
+        PORTAL_CATALOG: [
+            "getClientID",
+        ]
+    }
+    columns_to_remove = {
+
+    }
+
+    portal = api.get_portal()
+    for obj in portal.objectValues():
```

Comment:
Make it better explicit instead of iterating over all objects in the portal, e.g.:
```python
base_catalog_idxs = ["CreationDate", ]
for cat_id, idxs in indexes_to_remove.items():
    cat = api.get_tool(cat_id)
    if isinstance(cat, BaseCatalog):
        idxs.extend(base_catalog_idxs)
    ...
```